### PR TITLE
♻️ Attach firstElement and messageArray to assert errors

### PR DESCRIPTION
**Pull request by rcebulko**
Mirrors existing functionality in standalone assertion helpers

### CODE REVIEW

File: src/assert.js [hunk 1]
```diff
@@ -65,16 +74,25 @@ function assertion(errorCls, shouldBeTruthy, opt_message, var_args) {
...
+  const message = messageArgs.reduce(
+    (msg, subValue) => msg.replace('%s', subValue),
+    opt_message || 'Assertion failed'
+  );
+  const error = new errorCls(message);
+  error.messageArray = messageArgs;
+
+  // If an element is provided, add it to the error object
+  for (let i = 0; i < messageArgs.length; ++i) {
+    if (messageArgs[i]?.tagName) {
```
Comment (jridgewell) [inline src/assert.js:89]: Nit: This will be faster

```suggestion
    if (messageArgs[i]?.nodeType === 1) {
```

**Review by jridgewell — APPROVED** [review]

File: src/assert.js (hunk 1 above)
Comment (rcebulko) [reply src/assert.js]: Updated

File: src/assert.js [hunk 2]
```diff
@@ -65,16 +74,31 @@ function assertion(errorCls, shouldBeTruthy, opt_message, var_args) {
...
+      messageArray.push(subValue);
+    }
+
+    // If an element is provided, add it to the error object
+    if (!firstElement && subValue?.nodeType == 1) {
+      firstElement = subValue;
+    }
+
+    return subValue;
+  });
```
Comment (rcebulko) [inline src/assert.js:96]: @rsimha Do you know why this might thing these lines were not covered by tests? Every line is indeed covered

Comment (rsimha) [reply src/assert.js:96]: Not sure why it claimed zero coverage.

From CI logs, unit test coverage was successfully uploaded for the last commit in this PR: https://app.circleci.com/pipelines/github/ampproject/amphtml/2836/workflows/8eda1669-cac3-4370-a4f3-548652ffaf9f/jobs/32913/parallel-runs/0/steps/0-110

And codecov shows that the lines added by this PR are indeed covered (from running tests after this landed on `master`): https://codecov.io/gh/ampproject/amphtml/commit/ee8ac1edf18b76d00158ef949587b7e272fdad8a/ (link needs GH login)

Perhaps there was a blip the during PR build's coverage processing? Not sure. Can keep an eye, but can't think of any action items now.

Comment (rsimha) [reply src/assert.js:96]: Unless... you changed the code in one commit and added tests in a follow up commit, and these statuses are from the first commit that didn't have the coverage?

**Review by jridgewell — COMMENTED** [review]
Late approval.

**Pull request merged**